### ISSUE TITLE
Issue while compiling with AdamOptimizer

### ISSUE THREAD

**Issue by iamknownstranger** [tensorflow/tensorflow#33078]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): `model.compile(optimizer=tf.train.AdamOptimizer(),
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy']) `
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 Pro
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: None
- TensorFlow installed from (source or binary): Tensorflow -gpu from Anaconda
- TensorFlow version (use command below): 1.14.0
- Python version: 3.7
- CUDA/cuDNN version: 10
- GPU model and memory: Nvidia GeForce 1050Ti 4gb DDR5 VRAM

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
`
WARNING:tensorflow:From C:\ProgramData\Anaconda3\lib\site-packages\tensorflow\python\ops\init_ops.py:1251: calling VarianceScaling.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
WARNING:tensorflow:From convo.py:27: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.
**Describe the expected behavior**
Not expected warnings in the usage of AdamOptimizer. The code executed is working yet error is thrown on every execution
`
**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by hagianga21** [comment]
Have you tried :
`model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])`

**Comment by pandrey-fr** [comment]
As of 2.0, optimizers have been unified under the keras submodule; you therefore should use `tf.keras.optimizers.Adam` instead of the deprecated `tf.train.AdamOptimizer`. This is also what happens if you use `optimizer='adam'` as suggested by @hagianga21 although instantiating the optimizer yourself allows you to tweak its parameters.

**Comment by gadagashwini** [comment]
@iamknownstranger, Use `tf.compat.v1.train.AdamOptimizer` instead of `tf.train.AdamOptimizer`.
`tf.train.AdamOptimizer` is deprecated. Thanks!  
Please change the code as
`model.compile(optimizer=tf.compat.v1.train.AdamOptimizer(), loss='sparse_categorical_crossentropy', metrics=['accuracy'])`.
Thanks!

**Comment by gadagashwini** [comment]
@iamknownstranger, Is this still an issue. Thanks!
